snuba/query/query.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from deprecation import deprecated
+from typing import (
+    Any,
+    Iterable,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Sequence,
+    Tuple,
+    TypeVar,
+    Union,
+)
+
+
+Condition = Union[
+    Tuple[Any, Any, Any],
+    Sequence[Any],
+]
+
+Aggregation = Union[
+    Tuple[Any, Any, Any],
+    Iterable[Any],
+]
+
+Groupby = Iterable[Any]
```

Comment:
Not totally sure about this but I think `Aggregration` and `Groupby` should probably be `Sequence` and not `Iterable`.